app.rb

```diff
@@ -1,124 +1,175 @@
+    parent_permission = gets.chomp.downcase == 'y'
+    student = Student.new(name, age, parent_permission)
     @people << student
-    puts 'Student created successfully'
+    puts "#{student.name} has been added to the system.\n"
+    puts
   end
 
   def create_teacher
-    print 'Name:'
+    puts 'Creating a new teacher...'
+    puts 'Enter teacher name :'
     name = gets.chomp
-
-    print 'Age:'
+    puts 'Enter teacher age:'
     age = gets.chomp.to_i
-
-    print 'Specialization:'
+    puts 'Enter teacher specialization:'
     specialization = gets.chomp
-
-    @people << Teacher.new(age, specialization, name)
-    puts 'Teacher created successfully'
+    teacher = Teacher.new(name, age, specialization)
+    @people << teacher
+    puts "#{teacher.name} has been added to the system.\n"
+    puts
   end
 
   def create_book
-    print 'Title: '
+    puts 'Creating a new book...'
+    puts 'Enter book title:'
     title = gets.chomp
-
-    print 'Author: '
+    puts 'Enter book author:'
     author = gets.chomp
-
-    @books << Book.new(title, author)
-
-    puts 'Book created successfully'
+    book = Book.new(title, author)
+    @books << book
+    puts "#{book.title} has been added to the system.\n"
+    puts
   end
 
   def create_rental
-    puts 'Select a book from the following list by number'
-    @books.each_with_index do |book, index|
-      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
-    end
-
-    book_id = gets.chomp.to_i
+    book = select_book
+    person = select_person
+    start_date = select_start_date
+    rental = Rental.new(book, person, start_date)
+    @rentals << rental
+    puts "Rental created successfully\n"
+    puts
+  end
 
-    puts 'Select a person from the following list by number (not ID)'
-    @people.each_with_index do |person, index|
-      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  def select_book
+    puts 'Select a book from the following list (enter the corresponding number):'
+    @books.each.with_index(1) do |book, i|
+      puts "#{i}. #{book.title} by #{book.author}"
     end
+    book_choice = gets.chomp.to_i
+    book = @books[book_choice - 1]
+    if book.nil?
+      puts 'Invalid book selection.'
+      return select_book
+    end
+    book
+  end
 
-    person_id = gets.chomp.to_i
-
-    print 'Date: '
-    date = gets.chomp
+  def select_person
+    puts 'Select a person from the following list (enter the corresponding number):'
+    @people.each.with_index(1) do |person, i|
+      puts "#{i}. [#{person.class}] name: #{person.name}, ID: #{person.id}, age: #{person.age}"
+    end
+    person_choice = gets.chomp.to_i
+    person = @people[person_choice - 1]
+    if person.nil?
+      puts 'Invalid person selection.'
+      return select_person
+    end
+    person
+  end
 
-    @rentals << Rental.new(date, @people[person_id], @books[book_id])
-    puts 'Rental created successfully'
```

Comment:
- [ ] Please make sure that you separate pieces of logic into a separate class to make your code easier to understand as each core functionality has its own class. :blush:  This class you have here makes your code violates the Single Responsibility Principle (SRP) which states that **_every class, module, or function in a program should have one responsibility/purpose in a program. As a commonly used definition, "every class should have only one reason to change"_**. :+1: 
  
  The code above works perfectly but will lead to some challenges. We cannot make this code reusable for other classes or objects. The class has a whole lot of logic interconnected that we would have a hard time fixing errors. :relaxed: So please make sure that these logics of code are separated into separate classes :pray: You can read more about the [Best practices and Solid lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) and https://www.freecodecamp.org/news/solid-principles-single-responsibility-principle-explained/

Reply:
@Owusu-Desmond  Please, this is app.rb, and everything here has a single responsibility, which is creating the app.

Reply:
* As pointed out by the previous CR  I can see in your `App.rb` file some methods have not been refactored to comply with  the "Single-responsibility Principle". eg: the create student method should be responsible for only creating students HINT: you can extract all codes that ask for input or validation from users into another method so that for a method you can always pass all you need as arguments that way the code only has one responsibility.

* below 👇 are good examples of Single responsibility rules being followed
````ruby
def list_books
    @books.each do |book|
      puts book
    end
  end
````
````ruby
 def list_people
    @people.each do |person|
      puts person
    end
  end
````
* Kindly [follow this lesson for a reminder on how to refactor your code properly](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) and ensure `all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input`